core/src/main/scala/zio/oci/objectstorage/Live.scala

```diff
@@ -63,17 +63,25 @@ final class Live(unsafeClient: ObjectStorageAsyncClient) extends ObjectStorage.S
       ).map(r => ObjectStorageObjectListing.from(listing.namespace, listing.bucketName, r))
     }
 
-  override def getObject(namespace: String, bucket: String, name: String): ZStream[Blocking, BmcException, Byte] =
+  private def getObject(request: GetObjectRequest): ZStream[Blocking, BmcException, Byte] =
     ZStream
       .fromEffect(execute[GetObjectRequest, GetObjectResponse] { c =>
-        c.getObject(
-          GetObjectRequest.builder().namespaceName(namespace).bucketName(bucket).objectName(name).build(),
-          _: AsyncHandler[GetObjectRequest, GetObjectResponse]
-        )
+        c.getObject(request, _: AsyncHandler[GetObjectRequest, GetObjectResponse])
       })
       .flatMap(is => ZStream.fromInputStreamEffect(IO(is.getInputStream()).refineToOrDie[IOException]))
       .refineToOrDie[BmcException]
 
+  override def getObject(namespace: String, bucket: String, name: String, options: GetObjectOptions): ZStream[Blocking, BmcException, Byte] =
+    getObject(
+      GetObjectRequest
+        .builder()
+        .namespaceName(namespace)
+        .bucketName(bucket)
+        .objectName(name)
+        .range(options.range.map(r => new com.oracle.bmc.model.Range(r.startByte.map(Long.box).orNull, r.endByte.map(Long.box).orNull)).orNull)
```

Comment:
Oh you need to Long.box explicitly?

Reply:
Couldn't get it to compile otherwise, I'm sure there's a more graceful way to do the conversion but couldn't find it.